lib/puppet/face/help.rb

```diff
@@ -146,7 +146,9 @@ def all_application_summaries()
       if (is_face_app?(appname))
         begin
           face = Puppet::Face[appname, :current]
-          result << [appname, face.summary]
+          # Add deprecation message to summary if the face is deprecated
+          summary = face.respond_to?(:deprecated?) && face.deprecated? ? face.summary + _(" (Deprecated)") : face.summary
```

Comment:
Can we drop the `respond_to?` check necessary since https://github.com/puppetlabs/puppet/commit/7fefceae40d5ace8df2bd85aefa8bdee3246d9e1#diff-715723a3711cdfcf7cf7e524596200adR189?